spec/grape/validations/validators/default_spec.rb

```diff
@@ -298,4 +298,109 @@ def app
       end
     end
   end
+
+  context 'optional with nil as value' do
+    subject do
+      Class.new(Grape::API) do
+        default_format :json
+      end
+    end
+
+    def app
+      subject
+    end
+
+    context 'primitive types' do
+      [
+        [Integer, 0],
```

Comment:
Let's also test both a zero value, and a non-zero value for all number types, min/max for things like Date where available (e.g. `Time.at(0)`, and a non-empty String.